vlab-research/fly
# Feature/validator.js

### What changes have you made?
- created a `validator.js` module based on the `replybot` logic
-  applied replybot's `translateForm` function to translate the thankyou screens into the `fields` array for easier iteration
- set up validation checks for fields of type `short_text`, `number` and `thankyou_screen`
- created a new method on the `ResponseStore` class called `next` that evaluates a field value's validity and returns a next action command to the form
- expanded the unit tests for `validator.js` and `responseStore.js` to include the `validateFieldValue` and `next` logic
- included a new instance of `TypeError` that throws if the `validator` tries to evaluate a thankyou screen 

### Which issue(s) does this PR address?
#68 

### How to test
- `npm test` to check that all tests are passing
- `npm run dev` on `localhost:5000` and submit some test answers to the survey 
- when submitting an answer the form should either:
_a. allow the user to move to the next question
b. alert with an error message and prevent further navigation_
